sympy/core/tests/test_power.py

```diff
@@ -467,14 +468,26 @@ def test_issue_2993():
     assert str((-2.3*x - 2)**0.3) == '1.28386201800527*(-x - 0.869565217391304)**0.3'
     assert str((-2.3*x + 2)**0.3) == '1.28386201800527*(0.869565217391304 - x)**0.3'
     assert str((2.3*x + 2)**0.3) == '1.28386201800527*(x + 0.869565217391304)**0.3'
-    assert str((2.3*x - 4)**Rational(1, 3)) == '1.5874010519682*(0.575*x - 1)**(1/3)'
+    assert str((2.3*x - 4)**Rational(1, 3)) == '2**(2/3)*(0.575*x - 1)**(1/3)'
```

Comment:
I'm not sure if this is desired. Multiplying by 1.0 was meant to fix this, but it had the unintended consequence of losign digits. To preserve significant digits, I think we should multiply big by `Float(1, dps=max_dps)` where `max_dps` is the maximum number of digits in all of the floats found in the add.

Reply:
Neat suggestion! However I'm having trouble implementing what you suggest. Here's as far as I was able to make it:

```python
        elif e.is_Number and abs(e) != 1:
            # handle the Float case: (2.0 + 4*x)**e -> 2.**e*(1 + 2.0*x)**e
            c, m = zip(*[i.as_coeff_Mul() for i in self.args])
            # vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
            max_dps = -1
            if any(i.is_Float for i in c):  # XXX should this always be done?
                max_dps = max(prec_to_dps(i._prec), max_dps)
            # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            if any(i.is_Float for i in c):  # XXX should this always be done?
                big = -1
                for i in c:
                    if abs(i) >= big:
            # vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
                        big = Float(1,dps=max_dps)*abs(i)
            # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                if big > 0 and big != 1:
                    from sympy.functions.elementary.complexes import sign
                    bigs = (big, -big)
                    c = [sign(i) if i in bigs else i/big for i in c]
                    addpow = Add(*[c*m for c, m in zip(c, m)])**e
                    return big**e*addpow
```

I get the error

```
  File "............/add.py", line 393, in _eval_power
    max_dps = max(prec_to_dps(i._prec), max_dps)
AttributeError: 'Mul' object has no attribute '_prec'
```

Reply:
I would suggest to just keep it simple. If the largest number is a Rational then a Rational will get extracted. No harm done and the expression is more compact.

Reply:
otherwise, would this do it:
```python
elif e.is_Number and abs(e) != 1:
    # handle the Float case: (2.0 + 4*x)**e -> 2.**e*(1 + 2.0*x)**e
    c, m = zip(*[i.as_coeff_Mul() for i in self.args])
    if any(i.is_Float for i in c):  # XXX should this always be done?
        big = -1
        for i in c:
            if abs(i) >= big:
                big = abs(i)
        if big > 0 and big != 1:
            from sympy.functions.elementary.complexes import sign
            bigs = (big, -big)
            c = [sign(i) if i in bigs else i/big for i in c]
            addpow = Add(*[c*m for c, m in zip(c, m)])**e
            max_dps = -1
            for i in c:
                if i.is_Float:
                    max_dps = max(prec_to_dps(i._prec), max_dps)
            if max_dps != -1:
                big = Float(1, dps=max_dps)*big
            return big**e*addpow
```

Reply:
@smichr : Hmm... I get the following error when trying to run with your latest patch:

```
  File "/............/sympy/core/add.py", line 402, in _eval_power
    for i in c:
TypeError: 'Float' object is not iterable
```

Reply:
```
addpow = Add(*[d*m for d, m in zip(c, m)])**e
```

Reply:
@isuruf : Thanks that fixed the "not iterable" error. I had to update the two unit tests that were obviously affected. Please take a look and let us know if this seems acceptable. 

Thanks so much!